src/Bot.hs

```diff
@@ -203,12 +205,20 @@ builtinCommands =
     , ( "wiggle"
       , ( "Wiggle the tenticle (integration with https://github.com/tsoding/wiggle)"
         , transR (Identity . messageSender) $
-          cmapR (URI.encode . T.unpack . senderName) $
-          Reaction $ \(Identity name) -> do
-            request <- parseRequest [qms|http://localhost:8081/wiggle/{name}|]
-            void $ httpRequest request))
+          cmapR (URI.encode . T.unpack . senderName) wiggle))
     ]
 
+wiggle :: Reaction Identity String
+wiggle = Reaction $ \(Identity name) -> wiggle' name True
+  where
+    defaultTimeoutMillis = 10000
+    wiggle' name isRetry = do
+      request <- parseRequest [qms|http://localhost:8081/wiggle/{name}|]
+      response <- httpRequest request
+      let status = getResponseStatus response
+      when (status /= ok200) $
```

Comment:
Right now, when the server is down `httpRequest` doesn't return any status at all. It throughs an exception which short-circuit the entire Effect monad. See [BotState] for more info. To implement this feature we probably need to change the behaviour of `httpRequest` to make it possible to catch the HTTP exceptions.

[BotState]: https://github.com/tsoding/HyperNerd/blob/55d7a1baae12f11a0f30377f0dfe8f1b367d913a/src/BotState.hs#L82

Reply:
Yep, TODO is here. I think refactoring from `HttpRequest Request (Response B8.ByteString -> s)` to `HttpRequest Request (Maybe Response B8.ByteString -> s)` worth a separate issue. Everything else here seems to be ready.